include/beast/http/detail/chunk_encode.hpp

```diff
@@ -17,73 +19,109 @@ namespace beast {
 namespace http {
 namespace detail {
 
+struct chunk_extensions
+{
+    virtual ~chunk_extensions() = default;
+    virtual boost::asio::const_buffers_1 str() = 0;
+};
+    
+template<class ChunkExtensions>
+struct chunk_extensions_impl : chunk_extensions
+{
+    ChunkExtensions ext_;
+
+    chunk_extensions_impl(ChunkExtensions&& ext)
+        : ext_(std::move(ext))
+    {
+    }
+
+    chunk_extensions_impl(ChunkExtensions const& ext)
+        : ext_(ext)
+    {
+    }
+
+    virtual
+    boost::asio::const_buffers_1
+    str() override
+    {
+        auto const s = ext_.str();
+        return {s.data(), s.size()};
+    }
+};
+
+template<class T, class = void>
+struct is_chunk_extensions : std::false_type {};
+
+template<class T>
+struct is_chunk_extensions<T, beast::detail::void_t<decltype(
+    std::declval<string_view&>() = std::declval<T&>().str(),
+        (void)0)>> : std::true_type
```

Comment:
`void_t<expression,(void)0>` not necessary here, is it? Just:  `void_t<expression>` should suffice.